tools/goctl/api/gogen/main.tpl

```diff
@@ -3,17 +3,21 @@ package main
 import (
 	"flag"
 	"fmt"
-
+	
+	"github.com/num30/config"
```

Comment:
Why not use builtin conf?

Reply:
Here are some reasons:

- Ability to specify the env name when reading configuration . See [https://github.com/num30/config/pull/12](here)
- does not depend on file extension. Only need to specify filename without extension. Automatically looks for filename ending with .yml, .yaml, json and other. With previous go-zero strategy we have to specify the filename along with extensions explicitly. See https://github.com/num30/config#format
- Automatically reads variables from different sources: `config file<environment variables< command line`. For more see: https://github.com/num30/config#setting-configuration-values-construction_worker
- Supports validation of variables while reading. See https://github.com/num30/config#validations-underage

All the features I need are here. Go-zero conf loading package lacks the above features. Also

Reply:
but I'm really not willing to accept this PR with importing more dependencies.

Reply:
Why is there any problem? If there is already a clean solution available why not use it?

Reply:
I would have used it only for myself but there is no way to do it because struct tags required by this new library are different from what go-zero has.

Reply:
I have a workaround for this. 

before adding multiple struct tags
```go
type {
// A RpcServerConf is a rpc server config.
	RpcServerConf struct {
		service.ServiceConf
		ListenOn      string
		Etcd          discov.EtcdConf    `json:",optional"`
		Auth          bool               `json:",optional"`
		Redis         redis.RedisKeyConf `json:",optional"`
		StrictControl bool               `json:",optional"`
		// setting 0 means no timeout
		Timeout      int64 `json:",default=2000"`
		CpuThreshold int64 `json:",default=900,range=[0:1000]"`
		// grpc health check switch
		Health bool `json:",default=true"`
	}
}
```
after multiple struct tags

```go
type {
// A RpcServerConf is a rpc server config.
	RpcServerConf struct {
		service.ServiceConf `mapstructure:",squash"`
		ListenOn      string
		Etcd          discov.EtcdConf    `json:",optional"`
		Auth          bool               `json:",optional"`
		Redis         redis.RedisKeyConf `json:",optional"`
		StrictControl bool               `json:",optional"`
		// setting 0 means no timeout
		Timeout      int64 `json:",default=2000" default:"2000"`
		CpuThreshold int64 `json:",default=900,range=[0:1000]" default:"900" validate:"min=0,max=1000"` 
		// grpc health check switch
		Health bool `json:",default=true" default:"true"`
	}
}
```

this way go-zero does not have to add a new dependency but will allow the usage of https://github.com/num30/config if one wants by changing the main.go file

Reply:
what do you think?